A few changes to allow several --date on the CLI

oz opened the pull request:
There are a few comments attached to the commits on my fork. Feel free to dig in, if you're interested. Tests are green.

Commit:
I want to see more projects.

Commit:
Throw errors & pass spore-client to last cb

Commit:
Update to --date parameter

* Add -d alias for --date
* Allow several dates to be saved at once, while only requesting the
  task's metadata once (cheap copy):
  $ intervals.js -b -t 8 -d 2011-03-31 -d 2011-04-01 -d ...

francois2metz commented on intervals.js:
```diff
@@ -92,9 +92,11 @@ exports.addTime = function(options, client) {
                        projectsonly: 't'},
                       function(err, res) {
                           console.log('Choose client:');
+                          if (err) throw err;
```
Maybe we can improve error handling by calling callback parameter with the err instead of throwing the exception? Same for others lines.

oz replied:
Why not, error handling is a bit shaky in node, and JS, anyway... always end up by some code throwing up. Just keep the "limit: 40" :)

francois2metz replied:
On 01/04/2011 16:56, oz wrote:

> > Just keep the "limit: 40" :)

Yes, I doesn't have enough projets in timetask :).

francois2metz commented:
Thanks for the patch! 

I can probably remove the `Object.freeze()`. It is intended to prohibit change the contents of 'method' in middlewares. 
Do you have a stackstrace ? 
BTW, I will remove this line, or create a copy of the content.

francois2metz commented:
I have an error:

```
/home/francois/dev/timetask/bin/intervals.js:18
        dates    = (date.getDay) ? [date] : date,
                        ^
TypeError: Cannot read property 'getDay' of undefined
    at processTime (/home/francois/dev/timetask/bin/intervals.js:18:25)
    at /home/francois/dev/timetask/bin/intervals.js:63:26
    at /home/francois/dev/timetask/config.js:16:34
    at [object Object].<anonymous> (fs.js:86:5)
    at [object Object].emit (events.js:39:17)
    at afterRead (fs.js:843:12)
```

oz commented:
Oh I though optimist was supposed to set a default value when -d or --date were missing. Fixed that.

Commit:
Change options parsing of -d

I can't find a nice way of specifying it in optimist, and I guess it's
normal since 'aliasing' flags is just duplicating the values under
the two flags. Not what I want here.

oz commented:
On node-spore, I don't have a stack-trace: rather, I'll try to generate a test-case, and open the issue on the relevant project since it is not related to this one.
